python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -275,15 +309,19 @@ def _find_rail_stations(self, railStations,project_details,radiusinmeters,sample
 
         for idx, station in enumerate(railStations):
             crow_distance = self._haversine(lat, lon, station['Lat'], station['Lon'])
-            
+
             if crow_distance < (radiusinmeters / self.meters_per_mile):
-                walkDist = self._get_walking_distance(lat, lon, str(station['Lat']), str(station['Lon']))
-                walkDistMiles = walkDist / self.meters_per_mile
+                walkDist = self._get_walking_distance(lat, lon, str(station['Lat']), str(station['Lon']),db=db)
+                if type(walkDist) is not list:
```

Comment:
I'd recommend not having `self._get_walking_distance` return two different types of data in two different circumstances (cached or not). Instead, I'd recommend have _get_walking_distance always return a distance, and then this code can just  use the same append statment in line 321. This will make it more maintanable / easier to understand for future developers.